File: packages/javascript/lib/word/randClothing.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * What people wear, and what it is made of.
 *
 * `randWord({ theme: 'clothing' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randClothing({ language: 'ko', count: 3 }); // ['두루마기', '양말', '외투']
 * randClothing({ language: 'en', count: 3 }); // ['Cardigan', 'Mitten', 'Overcoat']
 */
export function randClothing(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Clothing words with the language and theme each one came from.
 *
 * @example
 * randClothing({ language: 'ko', output: 'detail' });
 * // [{ word: '두루마기', language: 'ko', theme: 'clothing' }]
 */
export function randClothing(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randClothing(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('clothing', options);
}

File: packages/dart/lib/src/word/rand_clothing.dart
import 'dart:math';

import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// What people wear, and what it is made of.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.clothing] to `randWordDetails` — Dart has
/// no overloads, and a second function per theme would be a second wall of them.
///
/// ```dart
/// randClothing(language: WordLanguage.ko, count: 3); // [두루마기, 양말, 외투]
/// randClothing(language: WordLanguage.en, count: 3); // [Cardigan, Mitten, Overcoat]
/// ```
List<String> randClothing({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  RandVocabulary vocabulary = RandVocabulary.full,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,

  /// Where the randomness comes from: `Random.secure()` for a value nobody may
  /// predict, `Random(42)` for one that has to come out the same every run.
  Random? random,
}) => randWord(
  language: language,
  theme: WordTheme.clothing,
  count: count,
  realism: realism,
  vocabulary: vocabulary,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
  random: random,
);
